core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -816,4 +877,53 @@ public void stateChanged(BufferState newState)
             }
         }
     }
+
+    private static final class DriverAndTaskTerminationTracker
+            implements Runnable
```

Comment:
instead of making this `Runnable` use `driverAndTaskTerminationTracker::driverDestroyed` in the caller

Reply:
Done.